src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -271,4 +302,27 @@ protected override bool TryGetExtensionsFromReference(AnalyzerReference referenc
             return false;
         }
     }
+
+    private record struct ProviderKey(string Language, TextDocumentKind DocumentKind, string DocumentExtension) : IEquatable<ProviderKey>
+    {
+        public bool Equals(ProviderKey other)
```

Comment:
Is the autogenerated equals/hashcode for records not appropriate here?

Reply:
it does not do case insensitive comparison of extensions